finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4DispatcherBase.scala

```diff
@@ -105,6 +105,14 @@ trait Netty4DispatcherBase[SendMsg <: Message, RecvMsg <: Message] {
         log.debug(e, "[%s] dispatcher closed", prefix)
         Future.exception(e)
 
+      case Throw(e: ChannelClosedException)
+        // if all streams have already been closed, then this just means that
+        // the client failed to send a GOAWAY frame...
+        if closed.get || streams.isEmpty =>
```

Comment:
Super minor, but I don't think you need to `closed.get` check here, since the previous case is `Throw(e) if closed.get`. I'd be inclined to write this as:

```
case Throw(e) if closed.get =>
case Throw(e: ChannelClosedException) if streams.isEmpty =>
case Throw(e) =>
```